platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -477,22 +479,61 @@ protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean
         return new LDAPConnectionFactory(host, port, lo);
     }
 
-    /**
-     * Added additional logging to the security LOGGER.
-     */
-    @Override
-    public boolean login() throws LoginException {
-        try {
-            boolean isLoggedIn = doLogin();
-
-            if (!isLoggedIn) {
-                SecurityLogger.audit("Username [" + user + "] failed LDAP authentication.");
+    void validateUsername(String username) throws LoginException {
+        boolean hasBadCharacters = false;
+        for (int i = 0; i < username.length(); i++) {
+            char curChar = username.charAt(i);
+            switch (curChar) {
+            case '\\':
+            case ',':
+            case '+':
+            case '"':
+            case '<':
+            case '>':
+            case ';':
+            case '#':
+                hasBadCharacters = true;
+                break;
             }
-            return isLoggedIn;
+            if (hasBadCharacters) {
+                throw new LoginException("Bad characters present in user name.");
+            }
+        }
+    }
 
-        } catch (Exception le) {
-            throw new LoginException("Username [" + user
-                    + "] could not log in successfuly using LDAP authentication due to an exception");
+    Set<Principal> getPrincipals() {
+        return ImmutableSet.copyOf(principals);
+    }
+
+    protected char[] getDecryptedPassword(String encryptedPassword) {
+
+        char[] decryptedPassword = null;
+        BundleContext bundleContext = getContext();
+
+        if (null != bundleContext) {
+            ServiceReference ref;
+            ref = bundleContext.getServiceReference(EncryptionService.class.getName());
```

Comment:
I agree with @AzGoalie.  It looks like we could have blueprint inject the EcryptionService and then we could inject a mock EcryptionService instead of overriding this method.

Here is an example.
https://github.com/codice/ddf/blob/17f05d54571d3b6e89146b990cc29d9a03fefb4e/platform/security/idp/security-idp-client/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L36